scene/3d/decal.cpp

```diff
@@ -30,6 +30,15 @@
 
 #include "decal.h"
 
+#include "scene/main/canvas_item.h"
+#include "scene/main/viewport.h"
+#include "scene/resources/animated_texture.h"
+#include "scene/resources/atlas_texture.h"
+#include "scene/resources/camera_texture.h"
+#include "scene/resources/mesh_texture.h"
+#include "scene/resources/texture.h"
+#include "scene/resources/texture_rd.h"
```

Comment:
It seems a bit unfortunate to add all these dependencies just for a warning, which will cause recompilation any time one of these texture types or their dependencies is edited.

Reply:
These includes could be debug-only at least.